### PR TITLE
feat: add nodeAgent middleware

### PR DESCRIPTION
**Pull request by judofyr**
Add support for using a single `http.Agent` for all requests in Node.js.

### CODE REVIEW

File: src/middleware.browser.ts [hunk 1]
```diff
@@ -16,3 +16,11 @@ export * from './middleware/proxy'
...
 export * from './middleware/urlEncoded'
 export type {RetryOptions} from './types'
+
+
+/**
+ * This middleware is a no-op in the browser.
+ */
+export function nodeAgent(opts?: any): any {
+  return {}
+}
```
Comment (stipsan) [inline src/middleware.browser.ts:26]: I'd prefer this to be in a `./middleware/agent/browser-agent` to keep in line with the convention already used for `keepAlive`.

File: src/middleware.ts [hunk 2]
```diff
@@ -9,6 +9,7 @@ export * from './middleware/jsonRequest'
 export * from './middleware/jsonResponse'
 export * from './middleware/keepAlive/node-keepAlive'
 export * from './middleware/mtls'
+export * from './middleware/node-agent'
```
Comment (stipsan) [inline src/middleware.ts:12]: I'd prefer this to be in a `./middleware/agent/node-agent` to keep in line with the convention already used for `keepAlive`.

**Comment by stipsan** [comment]
Note: the `Node.js current` test is likely going to fail here just as in #113, more info on why: https://github.com/sanity-io/get-it/pull/113#issuecomment-1513897794

File: src/middleware.ts (hunk 2 above)
Comment (judofyr) [reply src/middleware.ts:12]: Should it also be renamed to `agent`? That's also more consistent. However, I do like to explicitly highlight that this only does something in Node.js.

Comment (stipsan) [reply src/middleware.ts:12]: Yeah, the same is true for `keepAlive`, it doesn't do anything in Node.js.

I'm also fine with `keepAlive` being refactored to the same pattern you're using for `node-agent`, as it makes it much more immediately clear that it's a no-op. As long as both middleware's follow the same pattern 😄

**Comment by judofyr** [comment]
Wow, thanks for pointing me to the `keepAlive` middleware. It actually does something very similar. I've changed my implementation:

- `keepAlive` now _uses_ the `agent` middleware.
- The `agent` middleware now correctly handles HTTPS (using the previous logic in the `keepAlive` middleware).
- Unfortunately I ended up with some weird factory-pattern for the `keepAlive` since it now depends on the `agent`. Ideas for making it nicer?

**Review by stipsan — APPROVED** [review]
I'm fine with the factory pattern 👍

**Review by stipsan — APPROVED** [review]
All good on my end 👍  It does appear to be the case that the Node 14 error is related to a newer `vite` version being pulled in that no longer supports it (probably since it's past EOL). Pushing a fix to main and will rebase this branch afterwards, before merge and release :D

**Pull request merged**